Update and rename adding-payment-card-subscription-overview.md to Add…

zanyrenney opened the pull request:
…-a-payment-card-and-view-your-subcription.md

Added the images into the help site page and rename the title

https://github.com/Expensify/Expensify/issues/428744

<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$
PROPOSAL:


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->

</details>

zanyrenney committed:
Update and rename adding-payment-card-subscription-overview.md to Add-a-payment-card-and-view-your-subcription.md

Added the images into the help site page

dangrous reviewed (changes requested):
Couple of things:

* I think this needs the default header, e.g.
```
---
title: Subscription Management
description: How to manage your subscription
---
```
or whatever

* Also can we add some actual alt text for the images, instead of the placeholder `Insert alt text for accessibility here` haha? This is what would show on hover, or if they're using a screen reader.
* The deploy is getting the error `Articles have been renamed or deleted. Please add a redirect link for the old article links in redirects.csv` so we'll need to do that too

Thanks!

zanyrenney committed:
Update Add-a-payment-card-and-view-your-subcription.md

Changed the image placeholders and updated the Title from Dan's review.

zanyrenney committed:
Update redirects.csv

change the page link from: https://help.expensify.com/articles/new-expensify/billing-and-subscriptions/adding-payment-card-subscription-overview to https://help.expensify.com/articles/new-expensify/billing-and-subscriptions/add-a-payment-card-and-view-your-subcription

zanyrenney commented:
Okay @dangrous redirects also updated! thank you!

OSBotify commented:
A preview of your ExpensifyHelp changes have been deployed to https://b2c43a74.helpdot.pages.dev ⚡️
<!-- Created by actions-cool/maintain-one-comment -->

dangrous commented on docs/articles/new-expensify/billing-and-subscriptions/Add-a-payment-card-and-view-your-subcription.md:
```diff
@@ -1,4 +1,6 @@
-Subscription Management
+title: Subscription Management
+description: How to manage your subscription
+
```
```suggestion
---
title: Subscription Management
description: How to manage your subscription
---
```

Sorry! Needs these

dangrous reviewed (changes requested):
this looks good but one edit to the header and then I noticed that the url says `subcription` not `subscription` so that should be fixed in the name of the file AND in the redirects.

Everything else looks perfect though, sorry that I didn't catch that earlier!

zanyrenney commented:
Hmm this still doesn't look quite right @dangrous please don't merge yet 
![2024-10-11_10-07-40](https://github.com/user-attachments/assets/2ef5c49f-9fcf-4859-9839-3f43c50901a7)

zanyrenney commented:
I have a full k2 so gonna do this next week

dangrous commented:
no worries, I'm in no rush haha! As for that line accepting the suggestion [here](https://github.com/Expensify/App/pull/50567#discussion_r1796110537) should make that go away

zanyrenney committed:
Update docs/articles/new-expensify/billing-and-subscriptions/Add-a-payment-card-and-view-your-subscription.md

misspelled subscription!

Co-authored-by: Daniel Gale-Rosen <5487802+dangrous@users.noreply.github.com>

zanyrenney commented:
Okay @dangrous I accepted that suggestion and updated the title in there for my mistake to spell subscription right 🤪

zanyrenney commented:
Now let me give that redirect page another edit !

zanyrenney committed:
Update redirects.csv

misspelled subscription, updated in the file name and in the redirect.

zanyrenney commented:
Okay I just had to get this off my plate 😂 so I think i have now done it all @dangrous fingers crossed, let me know!

zanyrenney committed:
Update and rename Add-a-payment-card-and-view-your-subcription.md to Add-a-payment-card-and-view-your-subscription.md

updated to format line by line and title to be correct

zanyrenney commented:
OK okay this should be good now 😂 @dangrous

dangrous reviewed (approved):
yayyyy!

dangrous commented:
Oh wait nooooooooooo now there are conflicts, i was too slow!

can you click "Resolve conflicts" and then I think you should just be able to say "Accept both changes"

zanyrenney commented:
WELP. Sorry I am confused. I have:
- clicked resolve conflicts 
- am taken to this red/ yellow section but I don't see what I need to change? 
![2024-10-15_17-59-22](https://github.com/user-attachments/assets/9979a5f8-8ead-404e-b729-caebc4b202f4)
![2024-10-15_17-59-08](https://github.com/user-attachments/assets/9358a309-58b1-4c63-89dc-8080ae5d9ce3)

dangrous commented:
Okay I think the buttons I was talking about are from an extension I have. But still pretty easy. Basically in this case all you have to do is remove the weird looking lines. So:

```diff
- <<<<<<< zanyrenney-patch-1
https://help.expensify.com/articles/new-expensify/billing-and-subscriptions/adding-payment-card-subscription-overview,https://help.expensify.com/articles/new-expensify/billing-and-subscriptions/add-a-payment-card-and-view-your-subscription
- =======
https://help.expensify.com/articles/expensify-classic/articles/expensify-classic/expenses/Send-Receive-for-Invoices,https://help.expensify.com/articles/expensify-classic/articles/expensify-classic/expenses/Send-and-Receive-Payment-for-Invoices.md
https://help.expensify.com/articles/expensify-classic/articles/expensify-classic/expenses/Bulk-Upload-Multiple-Invoices,https://help.expensify.com/articles/expensify-classic/articles/expensify-classic/expenses/Add-Invoices-in-Bulk
- >>>>>>> main
```

So sorry this keeps dragging!

(The reason for this is the first bit between the weird lines is your change, and the second bit between the lines is a change that has been made by someone else while this was going on. Sometimes that can be a "conflict", like if there was: `variable = 3` and you changed it to `variable = 2` and someone else changed it to `variable = 1` at the same time. So you'd need to figure out which one you want to keep. In this case, it's basically saying that someone else added some things at the end, and you also added some things at the end. It doesn't know that it's totally fine for them to be in whatever order in this file, so it's asking you what you need to do. Which is just to keep everything.)

zanyrenney commented:
Hey @dangrous 

when you say remove the lines and then you pasted this: 
> - <<<<<<< zanyrenney-patch-1
https://help.expensify.com/articles/new-expensify/billing-and-subscriptions/adding-payment-card-subscription-overview,https://help.expensify.com/articles/new-expensify/billing-and-subscriptions/add-a-payment-card-and-view-your-subscription
- =======
https://help.expensify.com/articles/expensify-classic/articles/expensify-classic/expenses/Send-Receive-for-Invoices,https://help.expensify.com/articles/expensify-classic/articles/expensify-classic/expenses/Send-and-Receive-Payment-for-Invoices.md
https://help.expensify.com/articles/expensify-classic/articles/expensify-classic/expenses/Bulk-Upload-Multiple-Invoices,https://help.expensify.com/articles/expensify-classic/articles/expensify-classic/expenses/Add-Invoices-in-Bulk
- >>>>>>> main

Do you mean I remove that whole output and replace it?

I'm still not sure what you're saying I am afraid.

dangrous commented:
Yeah so literally delete just the three lines that were highlighted, so go from:

```
<<<<<<< zanyrenney-patch-1
https://help.expensify.com/articles/new-expensify/billing-and-subscriptions/adding-payment-card-subscription-overview,https://help.expensify.com/articles/new-expensify/billing-and-subscriptions/add-a-payment-card-and-view-your-subscription
=======
https://help.expensify.com/articles/expensify-classic/articles/expensify-classic/expenses/Send-Receive-for-Invoices,https://help.expensify.com/articles/expensify-classic/articles/expensify-classic/expenses/Send-and-Receive-Payment-for-Invoices.md
https://help.expensify.com/articles/expensify-classic/articles/expensify-classic/expenses/Bulk-Upload-Multiple-Invoices,https://help.expensify.com/articles/expensify-classic/articles/expensify-classic/expenses/Add-Invoices-in-Bulk
https://help.expensify.com/articles/expensify-classic/bank-accounts-and-payments/payments/Pay-Bills,https://help.expensify.com/articles/expensify-classic/bank-accounts-and-payments/payments/Create-and-Pay-Bills
>>>>>>> main
```
to
```
https://help.expensify.com/articles/new-expensify/billing-and-subscriptions/adding-payment-card-subscription-overview,https://help.expensify.com/articles/new-expensify/billing-and-subscriptions/add-a-payment-card-and-view-your-subscription
https://help.expensify.com/articles/expensify-classic/articles/expensify-classic/expenses/Send-Receive-for-Invoices,https://help.expensify.com/articles/expensify-classic/articles/expensify-classic/expenses/Send-and-Receive-Payment-for-Invoices.md
https://help.expensify.com/articles/expensify-classic/articles/expensify-classic/expenses/Bulk-Upload-Multiple-Invoices,https://help.expensify.com/articles/expensify-classic/articles/expensify-classic/expenses/Add-Invoices-in-Bulk
https://help.expensify.com/articles/expensify-classic/bank-accounts-and-payments/payments/Pay-Bills,https://help.expensify.com/articles/expensify-classic/bank-accounts-and-payments/payments/Create-and-Pay-Bills
```

zanyrenney reviewed (commented):
Dan updated the bullets, looks good!

zanyrenney commented:
I accepted his changes, now final change is on the redirects, doing that now.

zanyrenney commented:
Hey @dangrous those lines are no longer showing as 

<<<<<<< zanyrenney-patch-1
https://help.expensify.com/articles/new-expensify/billing-and-subscriptions/adding-payment-card-subscription-overview,https://help.expensify.com/articles/new-expensify/billing-and-subscriptions/add-a-payment-card-and-view-your-subscription
=======
https://help.expensify.com/articles/expensify-classic/articles/expensify-classic/expenses/Send-Receive-for-Invoices,https://help.expensify.com/articles/expensify-classic/articles/expensify-classic/expenses/Send-and-Receive-Payment-for-Invoices.md
https://help.expensify.com/articles/expensify-classic/articles/expensify-classic/expenses/Bulk-Upload-Multiple-Invoices,https://help.expensify.com/articles/expensify-classic/articles/expensify-classic/expenses/Add-Invoices-in-Bulk
https://help.expensify.com/articles/expensify-classic/bank-accounts-and-payments/payments/Pay-Bills,https://help.expensify.com/articles/expensify-classic/bank-accounts-and-payments/payments/Create-and-Pay-Bills
>>>>>>> main

They only show as

https://help.expensify.com/articles/new-expensify/billing-and-subscriptions/adding-payment-card-subscription-overview,https://help.expensify.com/articles/new-expensify/billing-and-subscriptions/add-a-payment-card-and-view-your-subscription
https://help.expensify.com/articles/expensify-classic/articles/expensify-classic/expenses/Send-Receive-for-Invoices,https://help.expensify.com/articles/expensify-classic/articles/expensify-classic/expenses/Send-and-Receive-Payment-for-Invoices.md
https://help.expensify.com/articles/expensify-classic/articles/expensify-classic/expenses/Bulk-Upload-Multiple-Invoices,https://help.expensify.com/articles/expensify-classic/articles/expensify-classic/expenses/Add-Invoices-in-Bulk
https://help.expensify.com/articles/expensify-classic/bank-accounts-and-payments/payments/Pay-Bills,https://help.expensify.com/articles/expensify-classic/bank-accounts-and-payments/payments/Create-and-Pay-Bills

Did you already make the changes? sorry I am so confused over this PR now.

dangrous reviewed (dismissed):
i... have no idea how that went away. But it works for me! Finally!

dangrous commented:
Argh wait no it's back - I think GitHub must have been confused? Can you try the same thing as above again? I also didn't see it but now I see it again

zanyrenney commented:
I can't see the lines at all. I am gonna schedule a sync as this is a bit insane.

dangrous commented:
Yeah I don't think it helps that GitHub seems to have some bugs. But yes, happy to sync! Let's get this over with haha

zanyrenney committed:
Merge branch 'main' into zanyrenney-patch-1
